File: ScalaCacheExample/src/test/scala/com/blogspot/miguelinlas3/scala/cache/CachedService.scala
package com.blogspot.miguelinlas3.scala.cache

case class SimpleObject(val x: Int , val y: Int)

class CachedService extends Caching {
  def buildSimpleObject (x: Int, y: Int): SimpleObject = {
    cache[SimpleObject](x + ":" + y) {
    	SimpleObject(x, y)
    }
  }
}

object CachedService {

  def main(args: Array[String]): Unit = {

    val cachedService = new CachedService()
    
    for (index <- 1 to 10){
      val obj = cachedService.buildSimpleObject(index, index)
      println("Result:" + obj.x + "--" + obj.y)
    }
    
    // all this elements should be cache hits
    for (index <- 1 to 10){
      val obj = cachedService.buildSimpleObject(index, index)
      println("Result:" + obj.x + "--" + obj.y)
    }
  }

}

File: AspectJCacheExample/src/test/java/com/blogspot/miguelinlas3/aspectj/cache/CachedService.java
package com.blogspot.miguelinlas3.aspectj.cache;

import com.blogspot.miguelinlas3.aspectj.cache.marker.Cachable;

public class CachedService {

	@Cachable(scriptKey="$x + ':' + $y")
	public Integer sum(Integer x, Integer y) {
		return x + y;
	}
	
	public static void main(String[] args) {
		CachedService cachedService = new CachedService();
		
		for(int i = 0 ; i <= 10 ; ++i){
			cachedService.sum(i, i);
		}
	}
}
